src/snowflake/snowpark/open_telemetry_prototype.py

```diff
@@ -0,0 +1,75 @@
+import inspect
+from typing import Dict, Union
+
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader, ConsoleMetricExporter
+
+
+resource = Resource(attributes={
+    SERVICE_NAME: "snowpark-python-open-telemetry"
+})
+# output to console for debug
+traceProvider = TracerProvider(resource=resource)
+processor = BatchSpanProcessor(ConsoleSpanExporter())
+traceProvider.add_span_processor(processor)
+trace.set_tracer_provider(traceProvider)
+
+tracer = trace.get_tracer("snowflake.snowpark.dataframe")
+
+
+
+def open_telemetry(name):
+    def open_telemetry_decorator(func):
+        def wrapper(*df, **params):
+            # get complete parameter list of the function
+            dataframe, parameters = parameter_decoder(df, params, func)
+            with tracer.start_as_current_span(name) as cur_span:
+                # store parameters in span
+                cur_span.set_attribute("function_parameters", str(parameters))
+
+                # store execution location in span
+                frame_info = inspect.stack()[-1]
+                cur_span.set_attribute("filepath", f"{frame_info.filename}#{frame_info.lineno}")
+
+                # stored method chain
+                cur_span.set_attribute("method chain", str(dataframe._plan.api_calls))
+                # collect query and query id after execution with query_history()
+                with dataframe._session.query_history() as query_listener:
+                    result = func(*df, **params)
+                    queries = []
+                    # remove query id for now as they can put it on the server side
+                    for query in query_listener.queries:
+                        queries.append({"dataframe.query.text": query.sql_text})
+                    cur_span.set_attribute("queries", str(queries))
```

Comment:
Is there any limit on the size of the attribute? cc @sfc-gh-bdrutu @sfc-gh-anxu 
The SQL text can be quite large right? I wonder what the behavior would be if we exceed the limit (if there are any), truncated text?

Reply:
After the discussion on Mar, 14, I think we decided to cut sql text for now